### PR TITLE
The issue where the index_col parameter of read_shapefile can only be index.

### PR DESCRIPTION
**Pull request by Anjieluo**
In previous code, the index_col parameter of the read_shapefile function in wntr.network.io seemed to only accept index (this required the element name in my shapefile to be index only). I wanted to use a field from my shapefile, such as id, to replace this index, instead of being restricted to using index. Therefore, I made some modifications and have provided some test documentation.
[test.zip](https://github.com/USEPA/WNTR/files/14920803/test.zip)

I hope to contribute to wntr, and I apologize if my idea is incorrect.

### CODE REVIEW

File: wntr/gis/network.py [hunk 1]
```diff
@@ -171,7 +171,7 @@ def _extract_geodataframe(df, crs=None, links_as_points=False):
         df = df_links[df_links['link_type'] == 'Valve']
         self.valves = _extract_geodataframe(df, crs, valves_as_points) 
         
-    def _create_wn(self, append=None):
+    def _create_wn(self, append=None, index_col='index'):
```
Comment (kbonney) [inline wntr/gis/network.py:174]: Adding `index_col` as an additional keyword argument is unnecessary here because the index name can be extracted from the dataframes.

File: wntr/gis/network.py [hunk 2]
```diff
@@ -192,7 +192,7 @@ def _create_wn(self, append=None):
             if element.shape[0] > 0:
                 assert (element['geometry'].geom_type).isin(['Point']).all()
                 df = element.reset_index()
-                df.rename(columns={'index':'name', 'geometry':'coordinates'}, inplace=True)
+                df.rename(columns={index_col:'name', 'geometry':'coordinates'}, inplace=True)
```
Comment (kbonney) [inline wntr/gis/network.py:195]: Here, rather than resetting the index and then renaming the new index column I think it is more straightforward to run `df = element.reset_index(names="name")` which will reset the index and rename the index column to `name` in one line.

File: wntr/network/io.py [hunk 3]
```diff
@@ -217,7 +217,7 @@ def from_dict(d: dict, append=None):
                     link["start_node_name"],
                     link["end_node_name"],
                     pump_type=pump_type,
-                    pump_parameter=link.setdefault("power")
+                    pump_parameter=link.setdefault("power",50.0)
```
Comment (kbonney) [inline wntr/network/io.py:220]: This change seems unrelated to the rest. Can you please remove this.

File: wntr/network/io.py [hunk 4]
```diff
@@ -350,7 +350,7 @@ def from_gis(gis_data, append=None):
     if isinstance(gis_data, dict):
         gis_data = WaterNetworkGIS(gis_data)
 
-    wn = gis_data._create_wn(append=append)
+    wn = gis_data._create_wn(append=append,index_col=index_col)
```
Comment (kbonney) [inline wntr/network/io.py:353]: This change causes an error because index_col is not defined in this context. It is also no longer necessary given the earlier suggestions.

**Review by kbonney — CHANGES_REQUESTED** [review]

**Comment by kbonney** [comment]
Thanks for the PR @Anjieluo. Please address the above comments.

**Comment by kbonney** [comment]
These updates will make #395 unnecessary.

File: wntr/gis/network.py (hunk 1 above)
Comment (Anjieluo) [reply wntr/gis/network.py:174]: Thank you for your reply, help me to be able to use WNTR better
